### DISCUSSION TITLE
Automation and orchestration for boost deployment and updates

### DISCUSSION THREAD

**Discussion by brendalee** [filecoin-project/boost#1171] [Ideas]
SPs want automatic deployment and update solutions for Boost. 

Example from SPs: 
- once you are hitting more then 1 setup it already becomes useful. And your miners, workers and other service are the same for your infra (Safely to assume). Pre-defining those settings specific to your infra should give you the ability to automate, orchestrate and update without manual intervention and by doing so saving cost on daily operational activities.
- In our case, it would be awesome to be able to replicate an existing setup. So instead of the automation of the first lotus node, miner and worker, I'm more thinking about deploying additional nodes, miners and workers based on existing ones. That would be very useful in the following use cases (for us):
  - redeploying an existing node because of crashes, hardware issues, etc.
  - deploying new workers when we get more hardware (more often than not, there are always tweaks that are missed)
  - deploying temporary workers

**Comment by brendalee** [comment]
Additional input from @hannahhoward 

Boost is currently a multi process model, and a lot of SPs may not turn on all these processes by default. It would be good to get a default configuration started (esp if local index directory is separated out) - this would help SPs get a clean install + startup with the multiple processes preconfigured (with default as on instead of off).

**Comment by jacobheun** [comment]
Going to transfer this to a discussion as we'll need to do more ideation and planning to get to a reasonable direction for implementation.
